### ISSUE TITLE
[Feature Request] Documentation for functions

### ISSUE THREAD

**Issue by streamich** [palantir/documentalist#72]
Is there a way to document just functions?

```ts
/**
 * This is my func.
 */
export const myFunc = () => {};
```

**Comment by giladgray** [comment]
here are the types it visits: https://github.com/palantir/documentalist/blob/master/src/plugins/typescript/visitor.ts#L44-L57. you can probably guess how to add functions.

**Comment by jamesmfriedman** [comment]
This library is basically my last only hope for generating types from TS at this point. I used to use react-docgen but that only works with Flow. I used Typedoc for a while, but its type declarations were INCREDIBLY complicated, and the JSON files up to 10 megs for a simple component...

My main issue is I'm not able to get any information from the Functional components in my docs. I really need the comment blocks that are associated with them... I poked around in the code but I don't know enough about it to find a way to include function declarations into the export JSON.

Any help would be great, I'm close to releasing my updated docs for https://jamesmfriedman.github.io/rmwc/ but this is a hurdle I still need to overcome.

**Comment by StJohn3D** [comment]
@jamesmfriedman one thing this library does really well is interfaces. If you declare interfaces for your functions then documenting them shouldn't be too hard.

**Comment by jamesmfriedman** [comment]
Thanks for the quick reply! Yeah I saw that as a workaround but VSCode and other tools read the main comment off the component itself. I might just go the route of having duplicate comments unless you know some other magical workaround.

**Comment by StJohn3D** [comment]
@jamesmfriedman Good point! I wonder how VsCode is doing it :)

**Comment by styu** [comment]
+1 to supporting function documentation, I’ve found myself needing to do
this for libraries that essentially just export functions and had to
duplicate comments just to get the docs to render

**Comment by adidahiya** [comment]
Fixed by #104
